packages/sandbox/src/lambda_function_log_streamer.ts

```diff
@@ -0,0 +1,169 @@
+import { functionOutputKey } from '@aws-amplify/backend-output-schemas';
+import { LogLevel, Printer } from '@aws-amplify/cli-core';
+import { BackendOutputClient } from '@aws-amplify/deployed-backend-client';
+import {
+  BackendIdentifierConversions,
+  TagName,
+} from '@aws-amplify/platform-core';
+import { BackendIdentifier, BackendOutput } from '@aws-amplify/plugin-types';
+import {
+  CloudFormationClient,
+  DescribeStacksCommand,
+} from '@aws-sdk/client-cloudformation';
+import { LambdaClient, ListTagsCommand } from '@aws-sdk/client-lambda';
+import { Arn, ArnFormat } from 'aws-cdk-lib';
+import { CloudWatchLogEventMonitor } from './cloudwatch_logs_monitor.js';
+
+/**
+ * Logs streamer for customer defined lambda functions in a sandbox.
+ */
+export class LambdaFunctionLogStreamer {
+  /**
+   * Creates an instance of LambdaFunctionLogStreamer
+   */
+  constructor(
+    private readonly lambda: LambdaClient,
+    private readonly cfnClient: CloudFormationClient,
+    private readonly logsMonitor: CloudWatchLogEventMonitor,
+    private readonly backendOutputClient: BackendOutputClient,
+    private readonly printer: Printer
+  ) {}
+
+  public setOutputLocation = (outputLocation: string) => {
+    this.logsMonitor.setOutputLocation(outputLocation);
+  };
+
+  /**
+   * Starts watching logs in the given sandbox.
+   * @param sandboxBackendId The sandbox backend identifier.
+   * @param functionNamesToFilter The function names to filter.
+   */
+  public startWatchingLogs = async (
+    sandboxBackendId: BackendIdentifier,
+    functionNamesToFilter?: string[]
+  ) => {
+    const backendOutput: BackendOutput =
+      await this.backendOutputClient.getOutput(sandboxBackendId);
+
+    const definedFunctionsPayload =
+      backendOutput[functionOutputKey]?.payload.definedFunctions;
+    const deployedFunctionNames = definedFunctionsPayload
+      ? (JSON.parse(definedFunctionsPayload) as string[])
+      : [];
+
+    // To use list-tags API we need to convert function name to function Arn since it only accepts ARN as input
+    const deployedFunctionNameToArnMap = await this.getFunctionArnFromNames(
+      sandboxBackendId,
+      deployedFunctionNames
+    );
+
+    if (!deployedFunctionNameToArnMap) {
+      this.printer.log(
+        `[Sandbox] Could not find any function in stack ${BackendIdentifierConversions.toStackName(
+          sandboxBackendId
+        )}. Streaming function logs will be turned off.`,
+        LogLevel.DEBUG
+      );
+      return;
+    }
+
+    for (const entry of deployedFunctionNameToArnMap) {
+      const listTagsResponse = await this.lambda.send(
+        new ListTagsCommand({
+          Resource: entry.arn,
+        })
+      );
+      const friendlyFunctionName =
```

Comment:
using `getBackendMetadata()` already does this filtering

Reply:
Do we get the tagged names there? I thought it was only the physical Arn

https://github.com/aws-amplify/amplify-backend/blob/57a5138dbbc92b385e2e46258fa165dfb422afe7/packages/deployed-backend-client/src/deployed_backend_client.ts#L346-L363

Reply:
ah, yeah if we need the friendly name, we'd need to do some extra processing or expand the interface the deployed backend client

Reply:
I don't think we are at a point that warrants to expand the interface of deployed backend client. Also one can argue that `tags` (or other resource specific properties) are not considered `metadata` and shouldn't be part of this package. We can look into adding the `friendly-name` to stack outputs to avoid `lambda:list-tags`, but I don't think we are there yet.